spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/actions/RewritePositionDeleteFilesSparkAction.java

```diff
@@ -418,16 +431,16 @@ private String jobDesc(RewritePositionDeletesGroup group, RewriteExecutionContex
   }
 
   static class RewriteExecutionContext {
-    private final Map<StructLike, Integer> numGroupsByPartition;
+    private final StructLikeMap<Integer> numGroupsByPartition;
     private final int totalGroupCount;
     private final Map<StructLike, Integer> partitionIndexMap;
     private final AtomicInteger groupIndex;
 
-    RewriteExecutionContext(
-        Map<StructLike, List<List<PositionDeletesScanTask>>> groupsByPartition) {
-      this.numGroupsByPartition =
-          groupsByPartition.entrySet().stream()
-              .collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue().size()));
+    RewriteExecutionContext(StructLikeMap<List<List<PositionDeletesScanTask>>> groupsByPartition) {
+      this.numGroupsByPartition = StructLikeMap.create(groupsByPartition.type());
```

Comment:
This would also become 1 line with `transformValues`:

```
this.numGroupsByPartition = groupsByPartition.transformValues(List::size);
```

If we decide not to add `transformValues`, can we add a helper method?